src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -168,15 +196,17 @@ export class UiSettingsClient implements IUiSettingsClient {
 
   async setMany(changes: Record<string, any>, scope?: UiSettingScope) {
     this.onWriteHook(changes, scope);
-
     if (scope) {
       await this.write({ changes, scope });
     } else {
       // group changes into different scope
-      const [global, personal] = this.groupChanges(changes);
+      const [global, personal, workspace] = this.groupChanges(changes);
       if (global && Object.keys(global).length > 0) {
         await this.write({ changes: global });
       }
+      if (workspace && Object.keys(workspace).length > 0) {
+        await this.write({ changes: workspace, scope: UiSettingScope.WORKSPACE });
+      }
```

Comment:
User may not be able to update workspace setting due to permission, shall we put workspace setting as the first write so that it won't partial success.